### PR TITLE
Added tests for Login and Register

### PR DESCRIPTION
**Pull request by maexled**
Created tests for Login and Register.

### CODE REVIEW

File: src/handlers/register_test.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+	if !ok {
+		t.Errorf("A valid registration should return a 200 status code")
+	}
+	if okResult.Payload.Name != username {
+		t.Errorf("Expected username %q but got %q", username, okResult.Payload.Name)
+	}
+
+	if okResult.Payload.ID != "user id" {
+		t.Errorf("Expected username %q but got %q", username, okResult.Payload.Name)
```
Comment (delvh) [inline src/handlers/register_test.go:81]:
```suggestion
		t.Errorf("Expected user ID %q but got %q", userID, okResult.Payload.ID)
```
(assuming `"user id"` gets extracted into `userID`)

File: src/handlers/login_test.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+	if !ok {
+		t.Errorf("A valid login should return a 200 status code")
+	}
+
+	if okResult.Payload.Name != username {
+		t.Errorf("Expected username %q but got %q", username, okResult.Payload.Name)
+	}
+
+	if okResult.Payload.ID != "user id" {
+		t.Errorf("Expected username %q but got %q", username, okResult.Payload.Name)
```
Comment (delvh) [inline src/handlers/login_test.go:72]:
```suggestion
		t.Errorf("Expected user ID %q but got %q", userID, okResult.Payload.ID)
```

File: src/handlers/login_test.go [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+	// and makes sure that all of them succeeded
+	defer ensure(t)
+
+	username := "Test User"
+	password := "password"
+	passwordHash := "$2a$10$KIKrid5AyyXHKHXRt.zS7OrlYWqYv2FUJOXVOktCotczFKRhmVBW."
+
+	expected := db.UserModel{
+		InnerUser: db.InnerUser{
+			ID:           "user id",
```
Comment (delvh) [inline src/handlers/login_test.go:29]: Same thing for the userID as below.

File: src/handlers/login_test.go [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+	mock.User.Expect(
+		// define your exact query as in your tested function
+		// call it with the exact arguments which you expect the function to be called with
+		// you can copy and paste this from your tested function, and just put specific values into the arguments
+		client.User.FindFirst(
+			db.User.Name.Equals(username),
+		),
+	).Returns(expected) // sets the object which should be returned in the function call
+
+	authenticator := auth.NewAuthenticator("asdf")
```
Comment (delvh) [inline src/handlers/login_test.go:47]: I think you should make that a global variable for this file given that it exists thrice.

File: src/handlers/login_test.go [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+	expected := db.UserModel{
+		InnerUser: db.InnerUser{
+			ID:           "user id",
+			CreatedAt:    time.Now(),
+			Name:         username,
+			Email:        "test@username.com",
+			PasswordHash: passwordHash,
+		},
+	}
```
Comment (delvh) [inline src/handlers/login_test.go:35]: If I see that correctly, this whole block can be global variables.

**Comment by robojones** [comment]
The flaky tests should be fixed with v1.16.1 of the prisma-client-go. @maexled can you run the tests a few times locally, to confirm?

File: src/handlers/login_test.go (hunk 4 above)
Comment (robojones) [reply src/handlers/login_test.go:48]: A thing to keep in mind here is, that in Go, global variables are editable. We can't make them `final` or something. This means that we could get side effects between the tests if the variable is accidentally modified in a test.

So generally, we should consider the benefits and drawbacks. I am not sure if we have any advantage from extracting global variables in our tests. I doubt that we will often change the values and, in this case, extracting the variables does not improve the readability.

Comment (robojones) [reply src/handlers/login_test.go:48]: Generally, I like the idea, that tests are self-contained. Meaning, that when a test fails, you can understand what the test does and what is happening, just by looking at the test – without jumping around or browsing through several files. What are your thoughts on this @delvh @maexled ?

**Comment by maexled** [comment]
Yes, after updating prisma-client-go to v1.16.1 the _TestRegisterUser_ShouldReturnOkIfUserNotAlreadyExists_ test passes (tried multiple times, everytime it passed). So the mistake was in the old mock of prisma-client-go

**Pull request merged**